src/main.js

```diff
@@ -43,40 +43,63 @@
 /* globals init */
 // @target "InDesign";
 
-(function(glob, app, undef) {
-
-  /**
-   * @class b
-   * @static
-   */
-  var pub = {};
-
-  /**
-   * The basil version
-   * @property VERSION {String}
-   * @cat Environment
-   */
-  pub.VERSION = "1.1.0";
-
-  // @include "includes/constants.js";
-  // @include "includes/public-vars.js";
-  // @include "includes/private-vars.js";
-  // @include "includes/global-functions.js";
-
-  // @include "includes/core.js";
-
-  // @include "includes/structure.js";
-  // @include "includes/environment.js";
-  // @include "includes/data.js";
-  // @include "includes/shape.js";
-  // @include "includes/color.js";
-  // @include "includes/typography.js";
-  // @include "includes/image.js";
-  // @include "includes/math.js";
-  // @include "includes/transformation.js";
-  // @include "includes/ui.js";
-
-  init();
-
-})(this, app);
 
+// clearing global space if it is still populated from previous run of a loop script
+// to ensure basil methods work properly
+if($.engineName === "loop" && $.global.basilGlobal) {
+  for (var i = basilGlobal.length - 1; i >= 0; i--) {
```

Comment:
shouldn't this  `basilGlobal[i]` not be `$.global.basilGlobal[i]` see my latest comment in discussion

Reply:
Ok, now I looked into this issue. I would leave all global variables as `basilGlobal` if possible. The *first* instance of `basilGlobal` (in the if expression) I cannot just leave as `basilGlobal`, because that throws an error:

```
basilGlobal ist undefiniert
```

That's why I put it as `$.global.basilGlobal` there. However, I probably could test for this instead:

```js
if($.engineName === "loop" && $.global.hasOwnProperty("basilGlobal") { ... }
```

Do you think that's clearer?
Still, the delete statement in the last line of this block needs to be

```js
delete $.global.basilGlobal;
```

as the `delete` statement can only delete properties.